ksqldb-rest-client/src/test/java/io/confluent/ksql/rest/client/KsqlRestClientTest.java

```diff
@@ -141,12 +153,85 @@ public void shouldParseHttpsAddressWithoutPort() throws Exception {
     assertThat(ksqlRestClient.getServerAddress(), is(serverURI));
   }
 
+  @Test
+  public void shouldIncludeAdditionalHeadersForCCloudConnectorRequest() throws Exception {
+    // Given:
+    final KsqlRestClient ksqlRestClient = clientWithServerAddresses(
+        SOME_SERVER_ADDRESS,
+        Optional.of(BasicCredentials.of(CCLOUD_API_KEY, CCLOUD_API_SECRET)));
+    ksqlRestClient.setIsCCloudServer(true);
+
+    final Map<String, String> expectedHeaders = ImmutableMap.of(
+        CCLOUD_CONNECT_USERNAME_HEADER, CCLOUD_API_KEY,
+        CCLOUD_CONNECT_PASSWORD_HEADER, CCLOUD_API_SECRET
+    );
+
+    // When:
+    ksqlRestClient.makeConnectorRequest("create connector ... ;", 0L);
+
+    // Then:
+    verify(client).target(new URI(SOME_SERVER_ADDRESS), expectedHeaders);
+    verify(target).postKsqlRequest("create connector ... ;", Collections.emptyMap(), Optional.of(0L));
+  }
+
+  @Test
+  public void shouldNotIncludeAdditionalHeadersForOnPremConnectorRequest() throws Exception {
+    // Given:
+    final KsqlRestClient ksqlRestClient = clientWithServerAddresses(SOME_SERVER_ADDRESS);
```

Comment:
Sure, I updated this test instead of adding a new one. I think we can be pretty confident that if setting `setIsCCloudServer(false)` causes additional credentials to not be sent even if they are provided, then no additional credentials will be sent if they are not provided.